beta-pipelines/multiome/atac.wdl

```diff
@@ -22,72 +23,129 @@ workflow ATAC {
-    fastq_gzipped_input_read2: "read 2 fastq file as input for the pipeline, the cellular barcodes must be the first part of the read name separated by colon"
+    read1_fastq_barcodes_gzipped: "read 1 fastq file as input for the pipeline, the cellular barcodes must be the first part of the read name seperated by colon"
+    read2_fastq_barcodes_gzipped: "read 2 fastq file as input for the pipeline, the cellular barcodes must be the first part of the read name separated by colon"
     output_base_name: "base name to be used for the pipelines output and intermediate files"
     monitoring_script : "script to monitor resource comsumption of tasks"
     tar_bwa_reference: "the pre built tar file containing the reference fasta and cooresponding reference files for the BWA aligner"
 
   }
 
+  call AddBarcodes {
+    input:
+      read1_fastq = read1_fastq_gzipped,
+      read3_fastq = read3_fastq_gzipped,
+      barcodes_fastq = read2_fastq_gzipped,
+      output_base_name = output_base_name
+  }
   call TrimAdapters {
     input:
-      fastq_input_read1 = fastq_gzipped_input_read1,
-      fastq_input_read2 = fastq_gzipped_input_read2,
+      fastq_input_read1 = AddBarcodes.fastq_barcodes_output_read1,
+      fastq_input_read3 = AddBarcodes.fastq_barcodes_output_read3,
       output_base_name = output_base_name,
       monitoring_script = monitoring_script
-    }
-
+   }
   call BWAPairedEndAlignment {
     input:
       fastq_input_read1 = TrimAdapters.fastq_trimmed_adapter_output_read1,
-      fastq_input_read2 = TrimAdapters.fastq_trimmed_adapter_output_read2,
+      fastq_input_read3 = TrimAdapters.fastq_trimmed_adapter_output_read3,
       tar_bwa_reference = tar_bwa_reference,
       output_base_name = output_base_name,
       monitoring_script = monitoring_script
-    }
+  }
     
-    output {
-      File bam_aligned_output = BWAPairedEndAlignment.bam_aligned_output
-   }   
+  output {
+    File bam_aligned_output = BWAPairedEndAlignment.bam_aligned_output
+  }   
 }
 
+  task AddBarcodes {
+    input {
+      File read1_fastq
+      File read3_fastq
+      File barcodes_fastq
+      String output_base_name
+      Int mem_size = 200
+      String docker_image = "us.gcr.io/broad-gotc-prod/atac_barcodes:1.0.3-1679503564"
+      Int disk_size = ceil(2 * ( size(read1_fastq, "GiB") + size(read3_fastq, "GiB") + size(barcodes_fastq, "GiB") )) + 200
+  }
+
+   parameter_meta {
+      output_base_name: "base name to be used for the output of the task"
+      docker_image: "the docker image using cutadapt to be used (default: )"
+      disk_size : "disk size used in trimming adapters step"
+  }
+      
+    # output names for trimmed reads
+    String fastq_barcodes_read1 = output_base_name + ".R1.barcodes.fastq"
+    String fastq_barcodes_read3 = output_base_name + ".R3.barcodes.fastq"
+   
+    # Adding barcodes to read 1 and read 3 fastq
+    command <<<
+      set -euo pipefail
+      mv ~{read1_fastq} r1.fastq.gz
```

Comment:
do we have to do these moves? can we just skip straight to gunzipping? so something like `gunzip ~{read1_fastq}` `gunzip ~{read3_fastq}`

Reply:
I think we'd have to grab the basenames of the files as a variable so that we could specify the unzipped file in the python command.